controllers/update_status.go

```diff
@@ -306,49 +305,44 @@ func optionList(options ...string) []string {
 // tryConnectionOptions attempts to connect with all the combinations of
 // versions and connection strings for this cluster and returns the set that
 // allow connecting to the cluster.
-func tryConnectionOptions(cluster *fdbv1beta2.FoundationDBCluster, r *FoundationDBClusterReconciler) (string, string, error) {
+func tryConnectionOptions(cluster *fdbv1beta2.FoundationDBCluster, r *FoundationDBClusterReconciler) (string, error) {
 	logger := log.WithValues("namespace", cluster.Namespace, "cluster", cluster.Name, "reconciler", "updateStatus")
-	versions := optionList(cluster.Status.RunningVersion, cluster.Spec.Version)
 	connectionStrings := optionList(cluster.Status.ConnectionString, cluster.Spec.SeedConnectionString)
 
-	originalVersion := cluster.Status.RunningVersion
 	originalConnectionString := cluster.Status.ConnectionString
 
-	if len(versions) == 1 && len(connectionStrings) == 1 {
-		return originalVersion, originalConnectionString, nil
+	if len(connectionStrings) == 1 {
+		return originalConnectionString, nil
 	}
 
-	logger.Info("Trying connection options",
-		"version", versions, "connectionString", connectionStrings)
+	logger.Info("Trying connection options", "connectionString", connectionStrings)
 
-	defer func() { cluster.Status.RunningVersion = originalVersion }()
 	defer func() { cluster.Status.ConnectionString = originalConnectionString }()
 
-	for _, version := range versions {
-		for _, connectionString := range connectionStrings {
-			logger.Info("Attempting to get connection string from cluster",
-				"version", version, "connectionString", connectionString)
-			cluster.Status.RunningVersion = version
-			cluster.Status.ConnectionString = connectionString
-			adminClient, clientErr := r.getDatabaseClientProvider().GetAdminClient(cluster, r)
+	for _, connectionString := range connectionStrings {
```

Comment:
1 at a time?

Reply:
We could run them in a go routine but in the current implementation of the admin client we would work over the list in a serial manner anyway (and I'm not sure how go routines and the FDB bindings will work out together). In general we should have two different connection strings only for a short amount of time.